src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableClient.cs

```diff
@@ -136,6 +137,22 @@ async Task<string> IDurableOrchestrationClient.StartNewAsync(string orchestrator
             return instance.InstanceId;
         }
 
+        private void GetStatusesNotToOverride(out OrchestrationStatus[] dedupeStatuses)
+        {
+            var overridableStates = this.config.Options.OverridableExistingInstanceStates;
+            if (overridableStates == OverridableStates.NonRunningStates)
+            {
+                dedupeStatuses = new OrchestrationStatus[]
+                {
+                    OrchestrationStatus.Running,
+                    OrchestrationStatus.ContinuedAsNew,
+                    OrchestrationStatus.Pending,
+                };
+            }
+
+            dedupeStatuses = null;
```

Comment:
While the Azure Storage implementation properly handles the null array case, it is possible that others will not. It might be a bit safer to use an empty array instead of null.

Reply:
Good point!